pkg/controller/workloaddispatcher/incrementaldispatcher_test.go

```diff
@@ -0,0 +1,289 @@
+		},
+		{
+			name:           "admission check nil",
+			dispatcherName: kueueconfig.MultiKueueDispatcherModeIncremental,
+			workload:       baseWorkloadBuilder.DeepCopy(),
+		},
+		{
+			name:           "admission check is rejected",
+			dispatcherName: kueueconfig.MultiKueueDispatcherModeIncremental,
+			workload:       baseWorkloadBuilder.DeepCopy(),
+			mkAcState: &kueue.AdmissionCheckState{
+				Name:  "ac1",
+				State: kueue.CheckStateRejected,
+			},
+		},
+		{
+			name:           "admission check is ready",
+			dispatcherName: kueueconfig.MultiKueueDispatcherModeIncremental,
+			workload:       baseWorkloadBuilder.DeepCopy(),
+			mkAcState: &kueue.AdmissionCheckState{
+				Name:  "ac1",
+				State: kueue.CheckStateReady,
+			},
+		},
+		{
+			name:           "already assigned to cluster",
+			dispatcherName: kueueconfig.MultiKueueDispatcherModeIncremental,
+			workload:       baseWorkloadBuilder.Clone().ClusterName("assigned").Obj(),
+			mkAcState: &kueue.AdmissionCheckState{
+				Name:  "ac1",
+				State: kueue.CheckStatePending,
+			},
+		},
+		{
+			name:           "workload is already finished",
+			dispatcherName: kueueconfig.MultiKueueDispatcherModeIncremental,
+			workload:       baseWorkloadBuilder.Clone().Finished().Obj(),
+			mkAcState: &kueue.AdmissionCheckState{
+				Name:  "ac1",
+				State: kueue.CheckStatePending,
+			},
+			remoteClusters: []string{"cluster1"},
+			clusters: []kueue.MultiKueueCluster{
+				*utiltesting.MakeMultiKueueCluster("cluster1").
+					KubeConfig(kueue.SecretLocationType, "cluster1").
+					Generation(1).
+					Obj(),
+			},
+		},
+		{
+			name:           "workload has quota reserved",
+			dispatcherName: kueueconfig.MultiKueueDispatcherModeIncremental,
+			workload:       baseWorkloadBuilder.DeepCopy(),
+			mkAcState: &kueue.AdmissionCheckState{
+				Name:  "ac1",
+				State: kueue.CheckStatePending,
+			},
+			remoteClusters: []string{"cluster1"},
+			clusters: []kueue.MultiKueueCluster{
+				*utiltesting.MakeMultiKueueCluster("cluster1").
+					KubeConfig(kueue.SecretLocationType, "cluster1").
+					Generation(1).
+					Obj(),
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			objs := []runtime.Object{}
+			if tc.mkAcState != nil {
+				tc.workload.Status.AdmissionChecks = []kueue.AdmissionCheckState{*tc.mkAcState}
+				ac := utiltesting.MakeAdmissionCheck(string(tc.mkAcState.Name)).
+					ControllerName(kueue.MultiKueueControllerName).
+					Parameters(kueue.GroupVersion.Group, "MultiKueueConfig", string(tc.mkAcState.Name)).
+					Obj()
+
+				objs = append(objs, ac)
+			}
+
+			if tc.workload != nil {
+				objs = append(objs, tc.workload)
+			}
+
+			if tc.mkAcState != nil {
+				mkConfig := utiltesting.MakeMultiKueueConfig(string(tc.mkAcState.Name)).Clusters("cluster1").Obj()
+				objs = append(objs, mkConfig)
+			}
+			scheme := runtime.NewScheme()
+			utilruntime.Must(kueue.AddToScheme(scheme))
```

Comment:
```suggestion
			if err := kueue.AddToScheme(scheme); err != nil {
				t.Fatalf(...)
			}
```

The `t.Fatalf` could help us rather than just panic.